src/Application/Processor/ProductLowestPriceBeforeDiscountProcessor.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\PriceHistoryPlugin\Application\Processor;
+
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingLogEntryInterface;
+use Sylius\PriceHistoryPlugin\Domain\Repository\ChannelPricingLogEntryRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class ProductLowestPriceBeforeDiscountProcessor implements ProductLowestPriceBeforeDiscountProcessorInterface
+{
+    private const PERIOD_IN_DAYS = 30;
+
+    public function __construct(private ChannelPricingLogEntryRepositoryInterface $channelPricingLogEntryRepository)
+    {
+    }
+
+    public function process(ChannelPricingInterface $channelPricing): void
+    {
+        if (!$this->isPromotionApplied($channelPricing)) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
+
+        $latestLogEntry = $this->getLatestLogEntry($channelPricing);
+        $lowestPriceInPeriod = $this->findLowestPriceInPeriod($latestLogEntry);
+
+        $channelPricing->setLowestPriceBeforeDiscount($lowestPriceInPeriod);
+    }
+
+    private function isPromotionApplied(ChannelPricingInterface $channelPricing): bool
+    {
+        return
+            $channelPricing->getOriginalPrice() !== null &&
+            $channelPricing->getPrice() < $channelPricing->getOriginalPrice()
+        ;
+    }
+
+    private function getLatestLogEntry(ChannelPricingInterface $channelPricing): ChannelPricingLogEntryInterface
+    {
+        $latestLogEntry = $this->channelPricingLogEntryRepository->findLatestOneByChannelPricing($channelPricing);
+        Assert::notNull($latestLogEntry);
```

Comment:
It should not be asserted. It is, I guess, a normal situation when all logs were removed by the `sylius:price-history:clear` command.

Reply:
As this processor is called once the product's price changed it always should be at least one logEntry. I added it to satisfy static analysis as the repository returns. I will change it to:
```php
        /** @var ChannelPricingLogEntryInterface $latestLogEntry */
        $latestLogEntry = $this->channelPricingLogEntryRepository->findLatestOneByChannelPricing($channelPricing);

        return $latestLogEntry;
```
wdyt?

Reply:
The processor could be called in various situations some of which maybe we can't even predict. For example, we run `sylius:price-history:clear` and then change `Channel::lowestPriceForDiscountedProductsCheckingPeriod`.

Reply:
Ok, I think I agree with you. It would be safer to simply set this price to null if, for some reason, no single log exists